web_app/api/vault.py

```diff
@@ -0,0 +1,49 @@
+"""
+Module for handling vault deposit operations in the SPOTNET API.
+This module provides endpoints for creating and managing vault deposits.
+"""
+
+import logging
+from decimal import Decimal
+from typing import Optional
+
+from fastapi import APIRouter, HTTPException
+
+from web_app.schemas.vault import VaultDepositRequest, VaultDepositResponse
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter(prefix="/api/vault", tags=["vault"])
+
+
+@router.post("/deposit", response_model=VaultDepositResponse)
+def deposit_to_vault(request: VaultDepositRequest) -> VaultDepositResponse:
+    """
+    Created a new vault deposit record.
+
+    Args:
+        request (VaultDepositRequest): The deposit request containing wallet_id, amount, and symbol
+
+    Returns:
+        VaultDepositResponse: The created deposit record with status.
+
+    Raises:
+        HTTPException: If there is an error processing the deposit.
+    """
+    try:
+        logger.info(f"Processing deposit request for wallet {request.wallet_id}")
+
+        # Return mock response for now since DB integration will come later
+        return VaultDepositResponse(
```

Comment:
Please, if you sync with main, why you didn't add DepositDBConnector here with appropriate crud method 
https://github.com/djeck1432/spotnet/blob/main/web_app/db/crud.py#L647C7-L647C25

Reply:
alright noted

Reply:
alright, it seems to be that I need to send code which you need to write, otherwise won't work